### ISSUE TITLE
`column-map` on three columns throws an exception

### ISSUE THREAD

**Issue by genmeblog** [techascent/tech.ml.dataset#356]
When `column-map` is applied on three columns exception is thrown:

```clojure
(let [ds (ds/->dataset {:foo (range 0 5)
                        :bar (repeatedly #(rand-int 100))
                        :baz (repeatedly #(rand-int 100))})]
  (ds/add-or-update-column ds :quz (apply col/column-map
                                          (fn [foo bar baz]
                                            (if (zero? (mod (+ foo bar baz) 7)) "mod 7" "not mod 7"))
                                          nil (ds/columns ds))))
```

```
1. Unhandled java.lang.RuntimeException
   Object cannot be casted to long: not mod 7

                Casts.java:   77  ham_fisted.Casts/longCast
                  emap.clj:   85  tech.v3.datatype.emap/emap-reader/reify
           LongBuffer.java:   57  tech.v3.datatype.LongBuffer/reduce
           Reductions.java:   82  ham_fisted.Reductions/serialReduction
          LongMutList.java:   54  ham_fisted.LongMutList/fillRange
             IMutList.java:  366  ham_fisted.IMutList/toLongArray
                   api.clj: 1946  ham-fisted.api/long-array-v
                   api.clj: 1943  ham-fisted.api/long-array-v
   copy_make_container.clj:   43  tech.v3.datatype.copy-make-container/eval20262/fn
              MultiFn.java:  244  clojure.lang.MultiFn/invoke
                  base.clj:  469  tech.v3.datatype.base/eval18766/fn
             protocols.clj:   92  tech.v3.datatype.protocols/eval16533/fn/G
          datatype_api.clj:  140  tech.v3.datatype-api/clone
          datatype_api.clj:  136  tech.v3.datatype-api/clone
              datatype.clj:  257  tech.v3.datatype/clone
              datatype.clj:  254  tech.v3.datatype/clone
                column.clj:  229  tech.v3.dataset.column/column-map
                column.clj:  200  tech.v3.dataset.column/column-map
               RestFn.java:  142  clojure.lang.RestFn/applyTo
                  core.clj:  671  clojure.core/apply
                  core.clj:  662  clojure.core/apply
```

But below works (two columns):

```clojure
(let [ds (ds/->dataset {:foo (range 0 5)
                        :bar (repeatedly #(rand-int 100))})]
  (ds/add-or-update-column ds :quz (apply col/column-map
                                          (fn [foo bar]
                                            (if (zero? (mod (+ foo bar) 7)) "mod 7" "not mod 7"))
                                          nil (ds/columns ds))))
;; => _unnamed [5 3]:
;;    | :foo | :bar |      :quz |
;;    |-----:|-----:|-----------|
;;    |    0 |   63 |     mod 7 |
;;    |    1 |   20 |     mod 7 |
;;    |    2 |   15 | not mod 7 |
;;    |    3 |   85 | not mod 7 |
;;    |    4 |   46 | not mod 7 |
```

**Comment by cnuernber** [comment]
This is now fixed and will work as expected in 7.000-beta-50 and up.

**Comment by otfrom** [comment]
thanks!
